### PR TITLE
Adds i_max option

### PR DESCRIPTION
**Pull request by emersion**
Replaces #4.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -1,7 +1,17 @@
-/** 
+/**
  *  PID Controller.
  */
 var Controller = function(k_p, k_i, k_d, dt) {
+  var i_max;
+  if (typeof k_p == 'object') {
```
Comment (Philmod) [inline lib/index.js]: `===`

File: lib/index.js [hunk 2]
```diff
@@ -1,7 +1,17 @@
...
+    var options = k_p;
+    k_p = options.k_p;
+    k_i = options.k_i;
+    k_d = options.k_d;
+    dt = options.dt;
+    i_max = options.i_max;
+  }
```
Comment (Philmod) [inline lib/index.js:13]: Can you update the Readme?

**Comment by Philmod** [comment]
Otherwise, it's :+1:

**Comment by emersion** [comment]
Done! :-P

File: README.md [hunk 3]
```diff
@@ -40,6 +45,11 @@ while (!goalReached) {
 }
 ```
 
+## Options
+* `k_p`, `k_i`, `k_d`: the PID's coefficients
+* `dt`: interval of time (in seconds) between two measures. If not provided, it will be automatically calculated.
+* `i_max`: the maximum absolute value of the integral term
```
Comment (Philmod) [inline README.md]: Can you mark this one as optional?

File: README.md [hunk 4]
```diff
@@ -9,13 +9,18 @@
...
 ## Example
-Let's take the example of a car cruise control.  We want the car driving at 120km/h.
+Let's take the example of a car cruise control. We want the car driving at 120km/h.
 
 ### Create a Controller instance
 `k_p`, `k_i` and `k_d` are the proportional, integral and derivative terms. `dt` is the interval of time between two measures. If not set, it will be automatically calculated.
 ```js
 var Controller = require('node-pid-controller');
-var ctr = new Controller(0.25, 0.01, 0.01, 1); // k_p, k_i, k_d, dt
+var ctr = new Controller({
```
Comment (Philmod) [inline README.md:24]: Can you show both options to pass the parameters?

**Comment by emersion** [comment]
Better like this?

**Pull request merged**